functions/Restore-DbaDatabase.ps1

```diff
@@ -583,7 +583,12 @@ function Restore-DbaDatabase {
                     $Completed = 'successfully'
                 }
                 catch {
-                    Write-Exception $_
+                    if ($_.FullyQualifiedErrorId -like 'Destination File*' ) {
+                        Stop-Function -Message "Restore of $databasename failed, $($_.FullyQualifiedErrorId) " -InnerErrorRecord $_.Exception.InnerException -silent:$silent    
```

Comment:
Catch like this instead:
```PowerShell
if ($_.CategoryInfo.Category -like "DeviceError") {
    Stop-Function -Message "Restore of $databasename failed, $_" -ErrorRecord $_ -Silent $silent
}
```
This allows you to reuse the original message same as before and you avoid parsing text to decide (which is what error categories are for).
See below on the proposed changes in `Restore-DBFromFilteredArray`